librz/core/disasm.c

```diff
@@ -1484,14 +1484,16 @@ static void ds_atabs_option(RDisasmState *ds) {
 		return;
 	}
 	int bufasm_len = rz_strbuf_length(&ds->asmop.buf_asm);
-	int size = bufasm_len * (ds->atabs + 1) * 4;
+	int size = bufasm_len * (ds->atabs + 1) * 4 + 4;
```

Comment:
the whole function for me is just dark magic... I have no idea what it does and it looks overly complicated... If you just want to fix the OOB write, ok. Rewriting the whole function with something understandable would be even better :D

Reply:
I would rewrite if I understood it :D